### PR TITLE
Extend RunJobTask with additional supported fields and document them

### PR DESCRIPTION
**Pull request by geekswaroop**
## Changes

Add additional fields that the upcoming RunJob task type supports:

- Allow configuring a job with a single runJob task
- Document run_job_task in docs/resources/job.md.
## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

**Review by alexott — CHANGES_REQUESTED** [review]
Looks good, but run `make fmt` on the code to fix the failing action.

**Review by nfx — CHANGES_REQUESTED** [review]
fields are not present in the OpenAPI spec. can't document them in TF before that.

File: docs/resources/job.md [hunk 1]
```diff
@@ -226,6 +226,11 @@ You can invoke Spark submit tasks only on new clusters. **In the `new_cluster` s
 
 You also need to include a `git_source` block to configure the repository that contains the dbt project.
 
+### run_job_task Configuration Block
+
+* `job_id` - (Required) ID of the job
+* `job_parameters` - (Optional) Job parameters for the task
```
Comment (alexott) [inline docs/resources/job.md:232]: We need to specify what type this and previous parameters are

**Review by alexott — CHANGES_REQUESTED** [review]

File: exporter/importables.go [hunk 2]
```diff
@@ -455,6 +455,20 @@ var resourcesMap map[string]importable = map[string]importable{
...
 					}
 				}
+				if task.RunJobTask != nil {
+					if task.RunJobTask.JobID != "" {
+						ic.Emit(&resource{
+							Resource: "databricks_sql_endpoint",
+							ID:       task.RunJobTask.JobID,
+						})
+					}
+					if task.RunJobTask.JobParameters != "" {
```
Comment (alexott) [inline exporter/importables.go:465]: This shouldn't be emitted at all.

File: jobs/resource_job.go [hunk 3]
```diff
@@ -98,6 +98,12 @@ type DbtTask struct {
 	WarehouseId       string   `json:"warehouse_id,omitempty"`
 }
 
+// RunJobTask contains information about RunJobTask
+type RunJobTask struct {
+	JobID         string `json:"job_id"`
+	JobParameters string `json:"job_parameters,omitempty"`
```
[the remark this answers is not in the document]
Comment (geekswaroop) [reply jobs/resource_job.go]: Yes, changed it to a map

File: docs/resources/job.md (hunk 1 above)
Comment (geekswaroop) [reply docs/resources/job.md]: Done

**Comment by geekswaroop** [comment]
@alexott @nfx this feature has been documented in OpenAPI [here](https://github.com/databricks/universe/pull/325746). All review comments have been addressed. Please review!

**Review by alexott — CHANGES_REQUESTED** [review]
can you please add a unit test for it? Like in this PR: https://github.com/databricks/terraform-provider-databricks/pull/2447/files

**Comment by geekswaroop** [comment]
@alexott added a test case and cleaned up branching.

File: jobs/resource_job.go [hunk 4]
```diff
@@ -249,6 +256,7 @@ type JobSettings struct {
 	PipelineTask           *PipelineTask       `json:"pipeline_task,omitempty" tf:"group:task_type"`
 	PythonWheelTask        *PythonWheelTask    `json:"python_wheel_task,omitempty" tf:"group:task_type"`
 	DbtTask                *DbtTask            `json:"dbt_task,omitempty" tf:"group:task_type"`
+	RunJobTask             *RunJobTask         `json:"run_job_task,omitempty" tf:"group:task_type"`
```
Comment (alexott) [inline jobs/resource_job.go:259]: Do we really support it in API 2.0 style?  I suggest not to do it as a top-level object and force use inside of  the `task` object

**Review by alexott — COMMENTED** [review]
looks good overall, but I'm really not sure if we need to support it as a top-level object

File: jobs/resource_job.go (hunk 4 above)
Comment (geekswaroop) [reply jobs/resource_job.go:259]: We should support RunJobTask in API 2.0 style as customers have several use cases of running a Single Task Job as a RunJobTask. cc: @gaborratky-db

Comment (alexott) [reply jobs/resource_job.go:259]: Right now we don't have parity between supported tasks - it should be discussed maybe in the office hours

Comment (gaborratky-db) [reply jobs/resource_job.go:259]: @alexott it turns out that customers can use Run Job task for various useful things:
 * Trigger the same job with different schedules and/or parameters configured in Run Job task jobs
 * Trigger a multi-task job with dependencies continuously from a continuous Run Job task job

Most customers are also familiar with the Jobs API 2.0 syntax so let's keep this supported.

File: jobs/resource_job.go [hunk 5]
```diff
@@ -191,6 +197,7 @@ type JobTaskSettings struct {
 	PythonWheelTask *PythonWheelTask `json:"python_wheel_task,omitempty" tf:"group:task_type"`
 	SqlTask         *SqlTask         `json:"sql_task,omitempty" tf:"group:task_type"`
 	DbtTask         *DbtTask         `json:"dbt_task,omitempty" tf:"group:task_type"`
+	RunJobTask      *RunJobTask      `json:"run_job_task,omitempty" tf:"group:task_type"`
```
Comment (nfx) [inline jobs/resource_job.go:200]: this is not part of the OpenAPI spec (https://docs.databricks.com/api/workspace/jobs/create) - we can't add anything that doesn't have public doc at the moment.

it's a hard requirement

**Review by nfx — DISMISSED** [review]
- PR is blocked until tje field is documented in https://docs.databricks.com/api/workspace/jobs/create
- we only support 2.1 APIs, no need to support 2.0 APIs

File: jobs/resource_job.go (hunk 5 above)
Comment (thaiphv) [reply jobs/resource_job.go:200]: The API doc has now supported it. Can this be unblocked?

File: docs/resources/job.md [hunk 6]
```diff
@@ -247,6 +247,11 @@ You can invoke Spark submit tasks only on new clusters. **In the `new_cluster` s
 
 You also need to include a `git_source` block to configure the repository that contains the dbt project.
 
+### run_job_task Configuration Block
+
+* `job_id` - (Required)(String) ID of the job
+* `job_parameters` - (Optional)(Map) Job parameters for the task
```
Comment (gaborratky-db) [inline docs/resources/job.md:253]: @geekswaroop let's not document Job parameters until the feature goes GA

Comment (geekswaroop) [reply docs/resources/job.md]: Done

**Review by alexott — APPROVED** [review]

**Comment by geekswaroop** [comment]
jenkins merge

**Review by mgyucht — APPROVED** [review]
LGTM

**Comment by gaborratky-db** [comment]
OpenAPI changes for run_job_task are already live, Databricks REST API reference has it documented.

**Pull request merged**